src/App.tsx

```diff
@@ -1,14 +1,63 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visibleTodos, setVisibleTodos] = useState<Todo[]>(todos);
+  const [isTodosLoaded, setIsTodosLoaded] = useState(true);
+  const [isUserLoaded, setIsUserLoaded] = useState(true);
+  const [todoId, setTodoId] = useState(0);
+  const [searchField, setSearchField] = useState('');
+  const [filteringMethod, setFilteringMethod] = useState('all');
+
+  useEffect(() => {
+    const getTodosFromServer = async () => {
+      const response = await getTodos();
+
+      setTodos(response);
+      setIsTodosLoaded(false);
+    };
+
+    getTodosFromServer();
+  }, []);
+
+  const findTodo = (id: number) => {
+    return todos.find(todo => todo.id === id) || todos[0];
+  };
+
+  useEffect(() => {
+    let todosCopy = [...todos];
+
+    if (filteringMethod === 'completed') {
+      todosCopy = todosCopy.filter(todo => {
+        return todo.completed;
+      });
+    }
+
+    if (filteringMethod === 'active') {
+      todosCopy = todosCopy.filter(todo => {
+        return !todo.completed;
+      });
+    }
```

Comment:
id you need to call some functionality according to the value of one variable, will be better to use switch / case